numpy/core/src/umath/ufunc_object.c

```diff
@@ -6384,11 +6416,25 @@ ufunc_at(PyUFuncObject *ufunc, PyObject *args)
         }
     }
     if (fast_path == 1) {
-    res = ufunc_at__fast_iter(ufunc, flags, iter, iter2, op1_array, op2_array,
-                                  strided_loop, &context, strides, auxdata);
+        /*
+         * Try to use trivial loop (1d, no casting, aligned) if
+         * - the matching info has a indexed loop
+         * - all operands are 1d
+         */
+        if (ufuncimpl->contiguous_indexed_loop != NULL) {
```

Comment:
Wouldn't it make sense to have the dimension tests for whether the contiguous loop can work be done here as well (i.e., call the function something like `ufunc_at__contigous_loop`)?

Or, alternatively, move this to `ufunc_at__fast_iter`?

It seems strange to have this "try_*" function in the middle of something that is quite declarative.

Reply:
An alternative might be to decide which path is taken already above (or in its own thing) and not name the function `try_` here.

One thing about the `get_...` thing: UFuncs in theory support including the cast in their inner-loop.  Not using a `get_...` function here makes it impossible to do both unless we start including the information that we are in an `at` loop in the resolver call.

(i.e. `resolve_descriptors` and `get_loop` don't strictly require a value array to be no-cast, they could return a different loop to implement `float64 + float32 -> float64` in theory, the single pointer here, cannot do that.)

Reply:
Let's see if there is demand to support that.